packages/odyssey-react/src/components/Link/index.tsx

```diff
@@ -41,7 +42,12 @@ export type Props = {
   /**
    * The human readable/percievable value shown to the user
    */
-  children: ReactText;
+  children: ReactNode;
+
+  /**
+   * The Icon to be inserted into the Link.
+   */
+  icon: keyof IconIndex;
```

Comment:
This should just be done via composition rather than a prop here yes? Also consider that we've now forced anyone who uses a `Link` to download every `Icon` component as well. That isn't ideal we should rework the API here